ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/MediaPlayerService.kt

```diff
@@ -799,6 +822,7 @@ class MediaPlayerService : Service() {
 
         mediaSession = MediaSessionCompat(applicationContext, "UltrasonicService")
         mediaSessionToken = mediaSession!!.sessionToken
+        mediaSessionEventDistributor.RaiseMediaSessionTokenCreatedEvent(mediaSessionToken!!)
```

Comment:
This solution is actually [suggested by Google](https://developer.android.com/reference/kotlin/android/support/v4/media/session/MediaSessionCompat)

> the owner of the session may pass its session token to other processes to allow them to create a MediaControllerCompat to interact with the session

That's why I think the MediaSession should be a thing of its own, which is used by the MediaBrowser and the MediaPlayer